src/hooks/useOnScreen.ts

```diff
@@ -0,0 +1,41 @@
+import { RefObject, useEffect, useRef, useState } from 'react';
+
+export const useOnScreen = (ref: RefObject<HTMLElement>) => {
```

Comment:
Nitpick: Should this be `useIsOnScreen`?

Reply:
yes nice catch - done!